File: Solution/java/legacycode/AssessmentServlet/src/main/java/com/tap/tflassessment/servlet/Services/SubjectService.java
package com.tap.tflassessment.servlet.Services;

import java.util.List;
import java.util.concurrent.CompletableFuture;

import com.tap.tflassessment.servlet.Entities.SubjectModel;

public interface SubjectService {

    public List<SubjectModel> getAllSubjects();

     public int addSubject(SubjectModel subject);
    
    public int deleteSubject(int subjectId);
}

File: Solution/dotnet/legacycode/AssessmentPortal/Microservices/APIforSubject/Services/SubjectService.cs
// using Transflower.Entities;
// using Transflower.SubjectAPI.Repositories.Interfaces;
// namespace Transflower.SubjectAPI.Services;

using APIforSubject.Entities;
using APIforSubject.Services.Interfaces;
using APIforSubject.Repositories.Interfaces;
namespace APIforSubject.Services;

public class SubjectService : ISubjectService
{
   private readonly ISubjectRepository _repository;

   public SubjectService(ISubjectRepository repository)
   {
      _repository = repository;

   }
   /*public async Task<SubjectModel> GetDetails(int subjectId)
   {
      return await _repository.GetSubjectById(subjectId);
   }*/
   public async Task<List<SubjectModel>> GetAllSubject()
   {
      return await _repository.GetAllSubject();
   }

   public async Task<int> AddSubject(SubjectModel subject)
   {
      return await _repository.AddSubject(subject);
   }

   public async Task<int> DeleteSubject(int subjectId)
   {
      return await _repository.DeleteSubject(subjectId);
   }


}